spec/integration/parser/compiler_spec.rb

```diff
@@ -522,4 +522,31 @@ class xport_test {
       expect(catalog).to have_resource("Notify[nbr2]").with_parameter(:message, 'overridden')
     end
   end
+
+
+  context "when converting catalog to resource" do
+    it "the same environment is used for compilation as for transformation to resource form" do
+        Puppet[:code] = <<-MANIFEST
+          notify { 'dummy':
+          }
+        MANIFEST
+
+        # Create an instance of a mocha state machine that checks that the environment is
+        # 'production'
+        in_production_state = Class.new do
+
+          def active?
+            Puppet.lookup(:current_environment).name == :production
+          end
+
+          def to_s
+            # displayed if the expectation about this state fails
+            "in production environment"
+          end
+        end.new
```

Comment:
@hlindberg nicely done, I forgot about mocha state machines. This is a bit magical as we're using a generic Class instead of a `Mocha::StateMachine::State` object. Can this be reworked to use the standard mocha APIs? Also I'm wondering if we should instead test that the `to_resource` takes place in non-default environment?